File: test_adzuna.py
#!/usr/bin/env python3
"""
Test script pour Adzuna API
Usage: python test_adzuna.py
"""

import requests
import json
from datetime import datetime

# Adzuna API credentials (gratuit: https://developer.adzuna.com/)
APP_ID = 'YOUR_APP_ID'  # Remplacer par ton app_id
APP_KEY = 'YOUR_APP_KEY'  # Remplacer par ton app_key

# Paramètres de recherche
params = {
    'country': 'fr',  # fr, us, uk, de, etc.
    'keywords': 'développeur react',
    'location': 'Paris',
    'results_per_page': 10,
    'page': 1
}

url = f"https://api.adzuna.com/v1/api/jobs/{params['country']}/search/{params['page']}"

query_params = {
    'app_id': APP_ID,
    'app_key': APP_KEY,
    'what': params['keywords'],
    'where': params['location'],
    'results_per_page': params['results_per_page'],
    'content-type': 'application/json'
}

print('🔍 Recherche Adzuna...')
print(f"URL: {url}")
print(f"Params: {params['keywords']} @ {params['location']}\n")

try:
    response = requests.get(url, params=query_params, timeout=10)
    response.raise_for_status()
    
    data = response.json()
    
    print('✅ Réponse reçue\n')
    print('📊 STATISTIQUES:')
    print('─' * 50)
    print(f"Total offres trouvées: {data.get('count', 0)}")
    print(f"Offres retournées: {len(data.get('results', []))}")
    print()
    
    if data.get('results'):
        print('📋 EXEMPLE D\'OFFRES:\n')
        
        for idx, job in enumerate(data['results'][:3], 1):
            print('\n' + '━' * 50)
            print(f"OFFRE #{idx}")
            print('━' * 50)
            print(f"📌 Titre: {job.get('title', 'N/A')}")
            print(f"🏢 Entreprise: {job.get('company', {}).get('display_name', 'N/A')}")
            print(f"📍 Localisation: {job.get('location', {}).get('display_name', 'N/A')}")
            
            salary_min = job.get('salary_min')
            salary_max = job.get('salary_max')
            if salary_min and salary_max:
                print(f"💰 Salaire: {salary_min:,.0f}€ - {salary_max:,.0f}€")
            else:
                print(f"💰 Salaire: Non spécifié")
            
            created = job.get('created')
            if created:
                date = datetime.fromisoformat(created.replace('Z', '+00:00'))
                print(f"📅 Date: {date.strftime('%d/%m/%Y')}")
            
            print(f"🔗 URL: {job.get('redirect_url', 'N/A')}")
            
            description = job.get('description', '')
            # Remove HTML tags
            import re
            clean_desc = re.sub('<[^<]+?>', '', description)
            print(f"📝 Description (extrait):")
            print(f"   {clean_desc[:200]}..." if clean_desc else "   N/A")
            
            print(f"🏷️  Catégorie: {job.get('category', {}).get('label', 'N/A')}")
            print(f"⏰ Type contrat: {job.get('contract_type', 'N/A')}")
        
        print('\n\n' + '━' * 50)
        print('📦 STRUCTURE COMPLÈTE D\'UNE OFFRE:')
        print('━' * 50 + '\n')
        print(json.dumps(data['results'][0], indent=2, ensure_ascii=False))
        
        print('\n\n' + '━' * 50)
        print('🔑 CHAMPS DISPONIBLES:')
        print('━' * 50)
        for field, value in data['results'][0].items():
            if isinstance(value, dict):
                subfields = ', '.join(value.keys())
                print(f"  • {field}: Object ({subfields})")
            else:
                print(f"  • {field}: {type(value).__name__}")
        
        # Statistiques supplémentaires
        print('\n\n' + '━' * 50)
        print('📈 ANALYSE DES DONNÉES:')
        print('━' * 50)
        
        # Salaires
        salaries = [j for j in data['results'] if j.get('salary_min') and j.get('salary_max')]
        if salaries:
            avg_min = sum(j['salary_min'] for j in salaries) / len(salaries)
            avg_max = sum(j['salary_max'] for j in salaries) / len(salaries)
            print(f"💰 Salaire moyen: {avg_min:,.0f}€ - {avg_max:,.0f}€")
            print(f"   ({len(salaries)}/{len(data['results'])} offres avec salaire)")
        
        # Entreprises
        companies = set(j.get('company', {}).get('display_name') for j in data['results'] if j.get('company'))
        print(f"🏢 Entreprises uniques: {len(companies)}")
        
        # Localisations
        locations = set(j.get('location', {}).get('display_name') for j in data['results'] if j.get('location'))
        print(f"📍 Localisations: {', '.join(list(locations)[:5])}")
        
        # Catégories
        categories = set(j.get('category', {}).get('label') for j in data['results'] if j.get('category'))
        print(f"🏷️  Catégories: {', '.join(list(categories)[:5])}")
        
    else:
        print('❌ Aucune offre trouvée')
        print('Réponse complète:', json.dumps(data, indent=2, ensure_ascii=False))
    
except requests.exceptions.RequestException as e:
    print(f'❌ Erreur requête: {e}')
except json.JSONDecodeError as e:
    print(f'❌ Erreur parsing JSON: {e}')
    print(f'Réponse brute: {response.text}')

print('\n\n💡 TIPS:')
print('─' * 50)
print('1. Créer un compte gratuit: https://developer.adzuna.com/')
print('2. Obtenir APP_ID et APP_KEY')
print('3. Remplacer les valeurs dans ce script')
print('4. Lancer: python test_adzuna.py')
print()
print('📚 Pays disponibles:')
print('  fr (France), us (USA), uk (UK), de (Allemagne),')
print('  ca (Canada), au (Australie), at (Autriche), etc.')
print()
print('🔧 Paramètres avancés disponibles:')
print('  • salary_min / salary_max: Filtrer par salaire')
print('  • full_time / part_time: Type de contrat')
print('  • permanent / contract: Durée')
print('  • sort_by: relevance, date, salary')
print()

File: test_adzuna.js
// Test script pour Adzuna API
// Usage: node test_adzuna.js

const https = require('https');

// Adzuna API credentials (gratuit: https://developer.adzuna.com/)
const APP_ID = ''; // Remplacer par ton app_id
const APP_KEY = ''; // Remplacer par ton app_key

// Paramètres de recherche
const params = {
  country: 'fr', // fr, us, uk, de, etc.
  keywords: 'développeur react', // Mots-clés
  location: 'Paris', // Ville
  results_per_page: 10, // Max 50
  page: 1
};

const url = `https://api.adzuna.com/v1/api/jobs/${params.country}/search/${params.page}?` +
  `app_id=${APP_ID}&` +
  `app_key=${APP_KEY}&` +
  `what=${encodeURIComponent(params.keywords)}&` +
  `where=${encodeURIComponent(params.location)}&` +
  `results_per_page=${params.results_per_page}&` +
  `content-type=application/json`;

console.log('🔍 Recherche Adzuna...');
console.log('URL:', url.replace(APP_KEY, '***'));
console.log('');

https.get(url, (res) => {
  let data = '';
  
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    try {
      const json = JSON.parse(data);
      
      console.log('✅ Réponse reçue\n');
      console.log('📊 STATISTIQUES:');
      console.log('─────────────────────────────────────');
      console.log(`Total offres trouvées: ${json.count || 0}`);
      console.log(`Offres retournées: ${json.results?.length || 0}`);
      console.log('');
      
      if (json.results && json.results.length > 0) {
        console.log('📋 EXEMPLE D\'OFFRES:\n');
        
        json.results.slice(0, 3).forEach((job, idx) => {
          console.log(`\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━`);
          console.log(`OFFRE #${idx + 1}`);
          console.log(`━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━`);
          console.log(`📌 Titre: ${job.title}`);
          console.log(`🏢 Entreprise: ${job.company?.display_name || 'N/A'}`);
          console.log(`📍 Localisation: ${job.location?.display_name || 'N/A'}`);
          console.log(`💰 Salaire: ${job.salary_min && job.salary_max ? 
            `${job.salary_min}€ - ${job.salary_max}€` : 'Non spécifié'}`);
          console.log(`📅 Date: ${job.created ? new Date(job.created).toLocaleDateString('fr-FR') : 'N/A'}`);
          console.log(`🔗 URL: ${job.redirect_url || 'N/A'}`);
          console.log(`📝 Description (extrait):`);
          console.log(job.description?.substring(0, 200).replace(/<[^>]*>/g, '') + '...' || 'N/A');
          console.log(`🏷️  Catégorie: ${job.category?.label || 'N/A'}`);
          console.log(`⏰ Type contrat: ${job.contract_type || 'N/A'}`);
        });
        
        console.log('\n\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━');
        console.log('📦 STRUCTURE COMPLÈTE D\'UNE OFFRE:');
        console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n');
        console.log(JSON.stringify(json.results[0], null, 2));
        
        console.log('\n\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━');
        console.log('🔑 CHAMPS DISPONIBLES:');
        console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━');
        const fields = Object.keys(json.results[0]);
        fields.forEach(field => {
          const value = json.results[0][field];
          const type = typeof value === 'object' && value !== null ? 
            `Object (${Object.keys(value).join(', ')})` : 
            typeof value;
          console.log(`  • ${field}: ${type}`);
        });
        
      } else {
        console.log('❌ Aucune offre trouvée');
        console.log('Réponse complète:', JSON.stringify(json, null, 2));
      }
      
    } catch (e) {
      console.error('❌ Erreur parsing JSON:', e.message);
      console.error('Réponse brute:', data);
    }
  });
  
}).on('error', (e) => {
  console.error('❌ Erreur requête:', e.message);
});

console.log('\n💡 TIPS:');
console.log('─────────────────────────────────────');
console.log('1. Créer un compte gratuit: https://developer.adzuna.com/');
console.log('2. Obtenir APP_ID et APP_KEY');
console.log('3. Remplacer les valeurs dans ce script');
console.log('4. Lancer: node test_adzuna.js');
console.log('');
console.log('📚 Pays disponibles:');
console.log('  fr (France), us (USA), uk (UK), de (Allemagne),');
console.log('  ca (Canada), au (Australie), at (Autriche), etc.');
console.log('');
